joomla/joomla-cms
# [com_joomlaupdate] Allow a different reinstall URL than the update URL

Pull Request for Issue #11488 
## Notice

Don't do that **never** on your Live Site please install a dev / test install!
#### Summary of Changes

With this patch applyed you can use a different reinstall URL than the update url.

For Updating you need just the `Joomla_3.6.x_to_3.6.2-Stable-Patch_Package.zip` package for reinstall you need the `Joomla_3.6.2-Stable-Update_Package.zip`.

So now if we have a reinstall URL configured lets use it.
#### Testing Instructions

Tip: Do not hit the `Update` or `Reinstall` button on testing else the applyed PR gets reverted!
- install 3.6.2
- go to com_joomlaupdate
- see the reinstall URL is `Joomla_3.6.2-Stable-Update_Package.zip`
- apply this patch
- switch the update server to http://www.jah-tz.de/downloads/core/list2.xml
- go to com_joomlaupdate again
- see the reinstall URL is `Joomla_3.6.x_to_3.6.2-Stable-Patch_Package.zip`
- switch back to the core Update server
- see that now we have  `Joomla_3.6.2-Stable-Update_Package.zip` back.
- switch the update server to http://www.jah-tz.de/downloads/core/list3.xml
- notice there is a update that points to `Joomla_3.6.2-Stable-Update_Package.zip`
#### Documentation Changes Required

There is a `core used only` new entry for the extension.xml used by the core updater.
#### Additional Information

The Joomla Core Update Server needs to be extended to use the function with this line. if that line is missing reinstallurl is the same as the update url:

```
<reinstallurl type="full" format="zip">https://github.com/joomla/joomla-cms/releases/download/3.6.2/Joomla_3.6.2-Stable-Update_Package.zip</reinstallurl>
```

A complete example:

```
<?xml version="1.0" ?>
<updates>
    <update>
        <name>Joomla! 3.6</name>
        <description>Joomla! 3.6 CMS</description>
        <element>joomla</element>
        <type>file</type>
        <version>3.6.2</version>
        <infourl title="Joomla!">https://www.joomla.org/announcements/release-news/5667-joomla-3-6-2-released.html</infourl>
        <downloads>
            <downloadurl type="full" format="zip">https://github.com/joomla/joomla-cms/releases/download/3.6.2/Joomla_3.6.x_to_3.6.2-Stable-Patch_Package.zip</downloadurl>
            <reinstallurl type="full" format="zip">https://github.com/joomla/joomla-cms/releases/download/3.6.2/Joomla_3.6.2-Stable-Update_Package.zip</reinstallurl>
        </downloads>
        <tags>
            <tag>stable</tag>
        </tags>
        <maintainer>Joomla! PLT</maintainer>
        <maintainerurl>https://www.joomla.org</maintainerurl>
        <section>STS</section>
        <targetplatform name="joomla" version="3.6" />
        <php_minimum>5.3.10</php_minimum>
    </update>
</updates>
```
